### ISSUE TITLE
remove functionality of Checker.setClassLoader and TreeWalker.setClassLoader

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#3773]
Problem is detected at #3750 .

functionality of `Checker.setClassLoader` and `TreeWalker.setClassLoader` should be completely removed from as it is required only for deprecated approach of class loading in deprecated JavadocMethodCheck implementation.

we can not remove it earlier: 
- it is causing validation problems at JavadocMethodCheck (failed UTs)
- it is used by eclipse-cs and maven plugin

So it is chekcstyle8 fix.

We can not easily re-implement JavadocMethodCheck due to non implemneted http://checkstyle.sourceforge.net/writingjavadocchecks.html#Boolean_flag_indicating_the_presence_of_unclosed_HTML_tags
without this flag we will never pass regression testing.

Methods will be completely removed once maven-checkstyle-plugin stops its usage of it.

**Comment by Kietzmann** [comment]
May I deal with it?

**Comment by romani** [comment]
@Kietzmann , you can try but beware it is not strait forward issue.

**Comment by rnveach** [comment]
> it is required only for deprecated approach of class loading in deprecated JavadocMethodCheck implementation

How are we handling this? Are we going to remove `JavadocMethodCheck` or wait for it to first be re-implemented?
No tests fail with it removed, but running Checkstyle on our sources produces the following violations:
````
execute:
     [echo] Checkstyle started (checkstyle_checks.xml): 03/08/2017 12:53:45 PM
[checkstyle] Running Checkstyle 8.2-SNAPSHOT on 938 files
[checkstyle] [ERROR] M:\checkstyleWorkspace\checkstyle\src\main\java\com\puppycrawl\tools\checkstyle\checks\annotation\AnnotationUseStyleCheck.java:262:8: Unable to get class information for @throws tag 'ConversionException'. [JavadocMethod]
[checkstyle] [ERROR] M:\checkstyleWorkspace\checkstyle\src\main\java\com\puppycrawl\tools\checkstyle\checks\annotation\AnnotationUseStyleCheck.java:272:8: Unable to get class information for @throws tag 'ConversionException'. [JavadocMethod]
[checkstyle] [ERROR] M:\checkstyleWorkspace\checkstyle\src\main\java\com\puppycrawl\tools\checkstyle\checks\annotation\AnnotationUseStyleCheck.java:282:8: Unable to get class information for @throws tag 'ConversionException'. [JavadocMethod]
[checkstyle] [ERROR] M:\checkstyleWorkspace\checkstyle\src\main\java\com\puppycrawl\tools\checkstyle\checks\blocks\EmptyCatchBlockCheck.java:155:8: Unable to get class information for @throws tag 'org.apache.commons.beanutils.ConversionException'. [JavadocMethod]
[checkstyle] [ERROR] M:\checkstyleWorkspace\checkstyle\src\main\java\com\puppycrawl\tools\checkstyle\checks\blocks\EmptyCatchBlockCheck.java:167:8: Unable to get class information for @throws tag 'org.apache.commons.beanutils.ConversionException'. [JavadocMethod]
[checkstyle] [ERROR] M:\checkstyleWorkspace\checkstyle\src\main\java\com\puppycrawl\tools\checkstyle\checks\coding\IllegalTokenTextCheck.java:140:8: Unable to get class information for @throws tag 'org.apache.commons.beanutils.ConversionException'. [JavadocMethod]
[checkstyle] [ERROR] M:\checkstyleWorkspace\checkstyle\src\main\java\com\puppycrawl\tools\checkstyle\checks\regexp\RegexpCheck.java:167:8: Unable to get class information for @throws tag 'org.apache.commons.beanutils.ConversionException'. [JavadocMethod]
[checkstyle] [ERROR] M:\checkstyleWorkspace\checkstyle\src\main\java\com\puppycrawl\tools\checkstyle\utils\CommonUtils.java:75:8: Unable to get class information for @throws tag 'ConversionException'. [JavadocMethod]
     [echo] Checkstyle finished (checkstyle_checks.xml) : 03/08/2017 12:55:05 PM
     [echo] Checkstyle started (checkstyle_non_main_files_checks.xml): 03/08/2017 12:53:45 PM
[checkstyle] Running Checkstyle 8.2-SNAPSHOT on 1292 files
     [echo] Checkstyle finished (checkstyle_non_main_files_checks.xml): 03/08/2017 12:55:05 PM
````

**Comment by romani** [comment]
> Are we going to remove JavadocMethodCheck or wait for it to first be re-implemented?

such Check is in use by users as it works on simple javadocs and it is enough now or users.
We can not remove without smth instead.

> No tests fail with it removed, but running Checkstyle on our sources produces the following violations

we could start introducing this regression, we will not support classpath for first new implementation of this Check. We are not type aware tool. All Checks should forget about such checks for classpath. May be one day in future we come back to this feature to become type-aware ... but not now.

**Comment by rnveach** [comment]
> we could start introducing this regression

So will we keep displaying these violations and suppress them in our configuration, or will we disable this violation completely in the code at https://github.com/checkstyle/checkstyle/blob/16f91dda1a15e0e3c3d7142a159f17db72c06d0f/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheck.java#L405-L409 ?

**Comment by romani** [comment]
@rnveach ,

`logLoadErrorImpl` is method from AbstractTypeAwareCheck.
All functionality of AbstractTypeAwareCheck should be removed from Checks completely even from old Checks - this does not work in a lot of cases.
We are in 8.X release, we can remove such class at any time with all related properties from Check(s).

**Comment by rnveach** [comment]
@romani Ok to close this issue?

**Comment by romani** [comment]
Please update title and description, as we actually keep method but made it do nothing.
Please create new issue and link it to maven plugin issue, to let us know when actually we can remove clasloader methods.

**Comment by rnveach** [comment]
Done.

**Comment by ingomohr** [comment]
Hi all!

We use Checkstyle for 10 years now, and we also built integrations based on its API.
Now, we want to migrate from Eclipse Checkstyle Plug-in 7.3.0 to the current version.
However, our integrations use the API that has been removed in 2019 w/ this issue.

Is there a chance to get it back?
This would save us weeks in migration. At the moment, the "missing" API is our blocker that prevents us from migrating.

Thanks in advance for your time!

Cheers
Ingo

**Comment by rnveach** [comment]
I also use this functionality, but sadly I don't think there is much chance of getting it back. Checkstyle is not a runtime analysis tool, but a static file analyzer. It does go against what checkstyle does, but it does expand on the useful checks one can build. You can build your own custom `Checker` and add the functionality back to it. I am not sure how easy it would be to pass it to all the FileSets and Checks though without just making it a static field and passing it. You still could not extend this to eclipse-cs or maven though.

@romani Share your thoughts. Is there anyway to bring this back without actually adding direct support?
